packages/contracts/contracts/core/plugin/PluginCloneable.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+import {PermissionManager} from "../permission/PermissionManager.sol";
+
+import {IDAO} from "../IDAO.sol";
+import {DaoAuthorizable} from "../component/DaoAuthorizable.sol";
+import {AppStorage} from "../../utils/AppStorage.sol";
+
+/// @title PluginClones
+/// @notice An abstract contract to inherit from when creating a proxy contract.
+/// This should be used to deploy EIP-1167 clones.
+abstract contract PluginCloneable is Initializable, ERC165, Context, AppStorage {
+    bytes4 public constant PLUGIN_INTERFACE_ID = type(PluginCloneable).interfaceId;
+
+    error ClonesInitAlreadyInitialized();
+
+    /// @notice used by the aragon to call by default for PluginCloneable interfaces.
+    /// @param _dao the dao address to set in a slot.
+    function __PluginCloneable_init(address _dao) external {
+        if (isInitialized()) {
+            revert ClonesInitAlreadyInitialized();
+        }
+        initialized();
+        setDAO(_dao);
+    }
```

Comment:
As pointed out before, `AppStorage` doesn't bring us any benefit for Minimal Clones. The dev has to remember to call `__PluginCloneable_init` and can forget this. With `AppStorage` this only gets more complicated and unintuitive. We should follow OZ standard practices and use their non-upgradeable `Initializable.sol` (which we are inheriting here and not using).

```suggestion
abstract contract PluginCloneable is ERC165, DaoAuthorizable {
    bytes4 public constant PLUGIN_CLONEABLE_INTERFACE_ID = type(PluginCloneable).interfaceId;

    /// @notice Initializes the plugin by storing the DAO contract.
    /// @param _dao The DAO address to be stored.
    function __PluginCloneable_init(IDAO _dao) external onlyInitializing {
        dao = _dao;
    }
```

Reply:
in this iteration we go with all plugins using AppStorage.

Reply:
What is the rationale behind this decision?

Reply:
so we can fully test appStorage and study it's pros and cons